lib/manageiq/appliance_console/database_replication.rb

```diff
@@ -115,25 +116,9 @@ def write_pgpass_file
     private
 
     def ask_for_cluster_database_credentials
-      self.database_name = just_ask("cluster database name", database_name)
-      self.database_user = just_ask("cluster database username", database_user)
-
-      count = 0
-      loop do
-        count += 1
-        password1 = ask_for_password("cluster database password", database_password)
-        # if they took the default, just bail
-        break if password1 == database_password
-        password2 = ask_for_password("cluster database password")
-        if password1 == password2
-          self.database_password = password1
-          break
-        elsif count > 1 # only reprompt password once
-          raise RuntimeError, "passwords did not match"
-        else
-          say("\nThe passwords did not match, please try again")
-        end
-      end
+      self.database_name     = just_ask("cluster database name", database_name)
+      self.database_user     = just_ask("cluster database username", database_user)
+      self.database_password = ask_for_new_password("cluster database password", :default => database_password, :allow_empty => true)
```

Comment:
NOTE I'm not sure if this was intentional but the old method did allow for an empty password

Reply:
That's weird - @jrafanie or @bdunne do you know?

Reply:
Or maybe because it has a default, allow empty allows it to use the default?

Reply:
https://github.com/ManageIQ/manageiq-appliance_console/pull/251/files#diff-58f7d485041385eb768bd8d25d7e83a11ccf1a1bbf2bc4fb136fcdcf9cd6b590L126 had a default as well but that checks for a 0 length password.  I think the default is only present if you've already configured a password but if this is the first time through there would be no default.

Reply:
Looks like it was like that back to the initial open source commit.

It had a concept of no password, which I assume means the default, likely already configured with a password.  I'm not sure though.


```diff
+    def ask_for_database_credentials
+      self.host     = ask_for_ip_or_hostname("database hostname or IP address", host) if host.blank? || !local?
+      self.database = just_ask("name of the database on #{host}", database) unless local?
+      self.username = just_ask("username", username) unless local?
+      count = 0
+      loop do
+        count += 1
+        password1   = ask_for_password_or_none("database password on #{host}", password)
+        # if they took the default, just bail
+        break if (password1 == password)
+        password2   = ask_for_password("database password again")
+        if password1 == password2
+          self.password = password1
+          break
+        elsif count > 1 # only reprompt password once
+          raise ArgumentError, "passwords did not match"
+        else
+          say("\nThe passwords did not match, please try again")
+        end
+      end
+    end
```

af467356d71d5c81990590933254cd54cde2c6b7

Reply:
Okay I think I'm going to keep the behavior the same as the internal database configuration and not allow a non-default empty password.

Reply:
@agrare I agree :+1: